### PR TITLE
feature: openpmd writer for 3d hockney solver

### PR DESCRIPTION
**Pull request by s-sajid-ali**
Draft PR to run CI during development.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -131,6 +131,25 @@ option(
...
+find_package(openPMD CONFIG)
+
+# if not found, fetch it
+if(NOT openPMD_FOUND)
+  include(FetchContent)
+  set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
+  set(openPMD_BUILD_CLI_TOOLS OFF)
+  set(openPMD_BUILD_EXAMPLES OFF)
+  set(openPMD_BUILD_TESTING OFF)
+  set(openPMD_INSTALL OFF)
```
Comment (ax3l) [inline CMakeLists.txt]: For the issue with
```suggestion
  set(openPMD_INSTALL ON)
```
I think the issue we see here might be due to the CMake version in CI, I cannot immediately see the same issue on my Ubuntu with CMake `3.23.2`. What is the CMake version in CI here?
```
CMake Error at /home/runner/work/synergia2/synergia2/build/_deps/openpmd-src/CMakeLists.txt:1080 (install):
  install EXPORT given no DESTINATION!


CMake Error at /home/runner/work/synergia2/synergia2/build/_deps/openpmd-src/CMakeLists.txt:1085 (install):
  install FILES given no DESTINATION!
```
https://github.com/openPMD/openPMD-api/blob/78baa4cc16b04b446787e69059f2b7b27aba136f/CMakeLists.txt#L1080-L1085
I will check more...

Comment (ax3l) [reply CMakeLists.txt:144]: For the issue with
```suggestion
  set(openPMD_INSTALL ON)
```
I think the issue we see here might be due to the CMake version in CI, I cannot immediately see the same issue on my Ubuntu with CMake `3.23.2`. What is the CMake version in CI here?
```
CMake Error at /home/runner/work/synergia2/synergia2/build/_deps/openpmd-src/CMakeLists.txt:1080 (install):
  install EXPORT given no DESTINATION!


CMake Error at /home/runner/work/synergia2/synergia2/build/_deps/openpmd-src/CMakeLists.txt:1085 (install):
  install FILES given no DESTINATION!
```
https://github.com/openPMD/openPMD-api/blob/78baa4cc16b04b446787e69059f2b7b27aba136f/CMakeLists.txt#L1080-L1085
I will check more...

File: CMakeLists.txt [hunk 2]
```diff
@@ -131,6 +131,25 @@ option(
...
 
+# OpenPMD
+find_package(openPMD CONFIG)
+
+# if not found, fetch it
+if(NOT openPMD_FOUND)
+  include(FetchContent)
+  set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
+  set(openPMD_BUILD_CLI_TOOLS OFF)
+  set(openPMD_BUILD_EXAMPLES OFF)
```
Comment (ax3l) [inline CMakeLists.txt]: Maybe more usable for the superbuild - and usable with `openPMD_INSTALL=OFF`, is:
```suggestion
  set(openPMD_BUILD_EXAMPLES OFF)
  set(openPMD_BUILD_SHARED_LIBS OFF)
```
that way you don't need to install and find a shared lib for openPMD.

Comment (ax3l) [reply CMakeLists.txt]: Doc update coming here: https://github.com/openPMD/openPMD-api/pull/1325

File: CMakeLists.txt (hunk 1 above)
Comment (ax3l) [reply CMakeLists.txt]: > What is the CMake version in CI here?

CI: CMake 3.22.1 per https://packages.ubuntu.com/jammy/cmake

Comment (ax3l) [reply CMakeLists.txt]: Fix coming via https://github.com/openPMD/openPMD-api/pull/1326

**Comment by s-sajid-ali** [comment]
Superseded by #105.

**Pull request closed without merging**